prebuild/Windows/preinstall.sh

```diff
@@ -0,0 +1,38 @@
+# expects node, VS, and MSYS environments to be set up already. does everything else.
+
+deps="cairo-2 png16-16 jpeg-8 pango-1.0-0 pangocairo-1.0-0 gobject-2.0-0 glib-2.0-0 turbojpeg gif-7 freetype-6 rsvg-2-2";
+
+# install cairo and tools to create .lib
+
+pacman --noconfirm -S \
+  wget \
+  unzip \
+  mingw64/mingw-w64-x86_64-binutils \
+  mingw64/mingw-w64-x86_64-tools \
+  mingw64/mingw-w64-x86_64-libjpeg-turbo \
+  mingw64/mingw-w64-x86_64-pango \
+  mingw64/mingw-w64-x86_64-cairo \
+  mingw64/mingw-w64-x86_64-giflib \
+  mingw64/mingw-w64-x86_64-freetype \
+  mingw64/mingw-w64-x86_64-fontconfig \
+  mingw64/mingw-w64-x86_64-librsvg \
+  mingw64/mingw-w64-x86_64-libxml2
```

Comment:
@chearon I had to add this libxml2 line. Do I also need to add something to line 3? (`deps=`)

Reply:
Yeah you do to get the `.lib` files. Did it work without doing that?

Reply:
Yeah, somehow that lib got in there. Output of the `tar -czvf` command from the logs in my last post (last line):

```
2020-05-01T04:06:38.2807995Z Release/
2020-05-01T04:06:38.2808850Z Release/libgdk_pixbuf-2.0.so.0
2020-05-01T04:06:38.3231600Z Release/libgthread-2.0.so.0
2020-05-01T04:06:38.3237935Z Release/libpango-1.0.so.0
2020-05-01T04:06:38.4053897Z Release/libz.so.1
2020-05-01T04:06:38.4099015Z Release/libgobject-2.0.so.0
2020-05-01T04:06:38.5040638Z Release/libharfbuzz.so.0
2020-05-01T04:06:38.8123409Z Release/libstdc++.so.6
2020-05-01T04:06:38.8662366Z Release/libgif.so.7
2020-05-01T04:06:38.8723402Z Release/libfontconfig.so.1
2020-05-01T04:06:38.9515542Z Release/libgmodule-2.0.so.0
2020-05-01T04:06:38.9525404Z Release/libpcre.so.1
2020-05-01T04:06:38.9962821Z Release/librsvg-2.so.2
2020-05-01T04:06:41.1066899Z Release/libpangocairo-1.0.so.0
2020-05-01T04:06:41.1218360Z Release/libexpat.so.1
2020-05-01T04:06:41.1531592Z Release/libgio-2.0.so.0
2020-05-01T04:06:41.7131079Z Release/.deps/
2020-05-01T04:06:41.7134142Z Release/.deps/Release/
2020-05-01T04:06:41.7136499Z Release/.deps/Release/canvas.node.d
2020-05-01T04:06:41.7139171Z Release/.deps/Release/obj.target/
2020-05-01T04:06:41.7141421Z Release/.deps/Release/obj.target/canvas.node.d
2020-05-01T04:06:41.7143937Z Release/.deps/Release/obj.target/canvas/
2020-05-01T04:06:41.7146407Z Release/.deps/Release/obj.target/canvas/src/
2020-05-01T04:06:41.7148812Z Release/.deps/Release/obj.target/canvas/src/bmp/
2020-05-01T04:06:41.7151117Z Release/.deps/Release/obj.target/canvas/src/bmp/BMPParser.o.d
2020-05-01T04:06:41.7153620Z Release/.deps/Release/obj.target/canvas/src/CanvasPattern.o.d
2020-05-01T04:06:41.7161364Z Release/.deps/Release/obj.target/canvas/src/Backends.o.d
2020-05-01T04:06:41.7165449Z Release/.deps/Release/obj.target/canvas/src/color.o.d
2020-05-01T04:06:41.7169159Z Release/.deps/Release/obj.target/canvas/src/init.o.d
2020-05-01T04:06:41.7181367Z Release/.deps/Release/obj.target/canvas/src/closure.o.d
2020-05-01T04:06:41.7185210Z Release/.deps/Release/obj.target/canvas/src/Image.o.d
2020-05-01T04:06:41.7189211Z Release/.deps/Release/obj.target/canvas/src/Canvas.o.d
2020-05-01T04:06:41.7192989Z Release/.deps/Release/obj.target/canvas/src/CanvasGradient.o.d
2020-05-01T04:06:41.7196962Z Release/.deps/Release/obj.target/canvas/src/backend/
2020-05-01T04:06:41.7200641Z Release/.deps/Release/obj.target/canvas/src/backend/ImageBackend.o.d
2020-05-01T04:06:41.7204171Z Release/.deps/Release/obj.target/canvas/src/backend/Backend.o.d
2020-05-01T04:06:41.7205217Z Release/.deps/Release/obj.target/canvas/src/backend/PdfBackend.o.d
2020-05-01T04:06:41.7208066Z Release/.deps/Release/obj.target/canvas/src/backend/SvgBackend.o.d
2020-05-01T04:06:41.7210907Z Release/.deps/Release/obj.target/canvas/src/register_font.o.d
2020-05-01T04:06:41.7213821Z Release/.deps/Release/obj.target/canvas/src/CanvasRenderingContext2d.o.d
2020-05-01T04:06:41.7216743Z Release/.deps/Release/obj.target/canvas/src/ImageData.o.d
2020-05-01T04:06:41.7219762Z Release/canvas.node
2020-05-01T04:06:41.7443529Z Release/libpixman-1.so.0
2020-05-01T04:06:42.0423568Z Release/libfreetype.so.6
2020-05-01T04:06:42.2641755Z Release/libcroco-0.6.so.3
2020-05-01T04:06:42.3323068Z Release/libcairo.so.2
2020-05-01T04:06:42.6984065Z Release/libfribidi.so.0
2020-05-01T04:06:42.7072388Z Release/libjpeg.so.62
2020-05-01T04:06:42.7332160Z Release/libpng16.so.16
2020-05-01T04:06:42.7949087Z Release/libglib-2.0.so.0
2020-05-01T04:06:43.0451484Z Release/libpangoft2-1.0.so.0
2020-05-01T04:06:43.0756996Z Release/libffi.so.6
2020-05-01T04:06:43.0767866Z Release/obj.target/
2020-05-01T04:06:43.0770916Z Release/obj.target/canvas.node
2020-05-01T04:06:43.1032178Z Release/obj.target/canvas/
2020-05-01T04:06:43.1038588Z Release/obj.target/canvas/src/
2020-05-01T04:06:43.1045185Z Release/obj.target/canvas/src/bmp/
2020-05-01T04:06:43.1050692Z Release/obj.target/canvas/src/bmp/BMPParser.o
2020-05-01T04:06:43.1056642Z Release/obj.target/canvas/src/closure.o
2020-05-01T04:06:43.1061819Z Release/obj.target/canvas/src/CanvasRenderingContext2d.o
2020-05-01T04:06:43.1166693Z Release/obj.target/canvas/src/ImageData.o
2020-05-01T04:06:43.1176701Z Release/obj.target/canvas/src/register_font.o
2020-05-01T04:06:43.1180824Z Release/obj.target/canvas/src/Image.o
2020-05-01T04:06:43.1207499Z Release/obj.target/canvas/src/CanvasPattern.o
2020-05-01T04:06:43.1225973Z Release/obj.target/canvas/src/backend/
2020-05-01T04:06:43.1229154Z Release/obj.target/canvas/src/backend/ImageBackend.o
2020-05-01T04:06:43.1232328Z Release/obj.target/canvas/src/backend/SvgBackend.o
2020-05-01T04:06:43.1241894Z Release/obj.target/canvas/src/backend/Backend.o
2020-05-01T04:06:43.1256586Z Release/obj.target/canvas/src/backend/PdfBackend.o
2020-05-01T04:06:43.1260209Z Release/obj.target/canvas/src/color.o
2020-05-01T04:06:43.1284377Z Release/obj.target/canvas/src/init.o
2020-05-01T04:06:43.1287528Z Release/obj.target/canvas/src/Backends.o
2020-05-01T04:06:43.1290889Z Release/obj.target/canvas/src/CanvasGradient.o
2020-05-01T04:06:43.1299270Z Release/obj.target/canvas/src/Canvas.o
2020-05-01T04:06:43.1369933Z Release/libxml2.so.2
```

Reply:
Oh duh, it's been a while since I've looked at this. You don't need the `.lib` file unless you're linking with a library. `libxml2` is included because _other_ libraries link to it. So it should actually _not_ be put in `deps=`.

Reply:
I posted the Ubuntu log instead of the Windows log. But yeah, looking at the Windows log, what you just said sounds correct!